etcdserver/api/v2http/client_auth.go

```diff
@@ -90,8 +90,7 @@ func hasKeyPrefixAccess(sec auth.Store, r *http.Request, key string, recursive b
 		plog.Warningf("auth: no such user: %s.", username)
 		return false
 	}
-	authAsUser := sec.CheckPassword(user, password)
-	if !authAsUser {
```

Comment:
@xiang90 So

```
err := bcrypt.CompareHashAndPassword([]byte("$2a$10$Zzmtqsix/KthzodN0t21iOHIUlejJeRESPVoH7LIFvc7RkyUK3zIG"), []byte(""))
fmt.Println(err == nil)
```

So empty password passes our `CheckPassword` function because it returns `nil` error.